src/ResourceManagement/Compute/EncryptionSettings.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Management.Compute.Fluent.Models;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Management.Compute.Fluent
+{
+    /// <summary>
+    /// Internal base type exposing settings to enable and disable disk encryption extension.
+    /// </summary>
+    public abstract partial class EncryptionSettings
+    {
+
+        /// <return>Encryption extension protected settings.</return>
+        abstract internal DiskEncryptionSettings StorageProfileEncryptionSettings();
```

Comment:
Either with prefix of `Get` or provide property with get accessor.

Reply:
Done